File: API_R_01Agosto2024/XamaCardoso/q3_cidades.py
def main():
    count = 1;
    n_cidades = int(input("Quantas cidades deseja computar? "));
    somarenda = 0;
    maior_renda = -9999999999999;
    menor_renda = 9999999999999;
    while (count <= n_cidades):
        renda = float(input(f"Renda da cidade {count} (R$): "));
        if renda != 0:
            if renda > maior_renda:
                maior_renda = renda;
            if renda < menor_renda:
                menor_renda = renda;

            somarenda += renda;
            count+=1;
            continue;
        
        break;
    
    if count > 1:
        print(f"""

    =============== RESULTADO ==================          
    N° de cidades: {count-1}
    Renda média  : R$ {(somarenda/(count-1)):.2f}
    Maior renda  : R$ {maior_renda:.2f}
    Menor renda  : R$ {menor_renda:.2f}
""");


main();

File: API_R_01Agosto2024/Nicolas Rafael Silva Alves/API-Recuperação/q3_cidades.js
import { ask, getNumberPositive, getNumberPositiveZero } from "./Funções/io.js"; // Importação das funções de IO

function main() {
    console.clear();

    //Apresentação
    let message = '\n[-----------Renda Per Capita-----------]\nDigite 0 renda para finalizar a entrada!\n'
    console.log(message);

    //Entrada
    let numCidades = getNumberPositive("Quantas cidades serão informadas: ");

    //Processamento
    let totalCidades = numCidades
    let rendaAtual = 1
    let maiorRenda = 0;
    let menorRenda = 0;
    let somaRenda = 0;
    let index = 1

    while (totalCidades > 0 && rendaAtual != 0){

        rendaAtual = getNumberPositiveZero(`Digite a renda da cidade ${index}: `);
        index++

        if (totalCidades == numCidades){
            menorRenda = rendaAtual
        }

        totalCidades -= 1
        
        somaRenda += rendaAtual;

        if (rendaAtual > maiorRenda){
            maiorRenda = rendaAtual;
        }
        else if (rendaAtual < menorRenda){
            menorRenda = rendaAtual;
        }
    }

    let rendaMedia = somaRenda/numCidades;

    //Saída
    console.clear();

    message = (`
    ============================================
     • Renda Média: R$ ${rendaMedia.toFixed(2)} 
     • Menor Renda: R$ ${menorRenda.toFixed(2)}
     • Maior Renda: R$ ${maiorRenda.toFixed(2)} 
    ============================================\n`);
    
    console.log(message);
    
    // Mensagem para limpar console apos finalizar programa
    if (! ask("Finalizar Programa?(Press Enter)"));
        console.clear();
}

main();